native_client/deepspeech.cc

```diff
@@ -666,10 +744,28 @@ DS_SpeechToText(ModelState* aCtx,
   if (status != DS_ERR_OK) {
     return nullptr;
   }
+      
   DS_FeedAudioContent(ctx, aBuffer, aBufferSize);
   return DS_FinishStream(ctx);
 }
 
+Metadata*
+DS_SpeechToTextExtended(ModelState* aCtx,
+                const short* aBuffer,
+                unsigned int aBufferSize,
+                unsigned int aSampleRate)
+{
+  StreamingState* ctx;
```

Comment:
The whole body should be factorized with `DS_SpeechToText`, it's doing the same except for the finish call